### PR TITLE
Move to mkdocs

### PR DESCRIPTION
**Pull request by maebert**
This PR moves the documentation to [MkDocs](https://www.mkdocs.org). This has several advantages:

- Significantly easier and faster build process than sphinx
- Docs are written in Markdown instead of ReST, which lowers the barrier to entry for new contributors
- Direct deployment to github-pages

### Development

```sh
pip install mkdocs
make html # opens the docs and live-reloads
make docs # builds docs and deploys them to gh-pages
```

### This PR also

- [x] Consolidates doc edits from master, rc1, and rc2
- [x] Simplifies and modernizes the landing page
    landing
- [x] Modernizes the assets
- [x] Includes wonderful social banners for sharing on Twitter, Facebook, etc:
    image
- [x] Switches to a modified ReadTheDocs theme that fixes some readability issues (turns out the web has changed quite a bit since 2012...):
    docs

### CODE REVIEW

**Comment by maebert** [comment]
@wren @micahellison Anything blocking this merge? It doesn't touch functionality, but it consolidates a lot of doc edits from various branches to get a clean slate.

**Comment by wren** [comment]
@maebert Nope, I think this is a good idea. It's only held up by organizing the backlog, which has been my priority so far. 

Although, I'm just a little confused about why jrnl.sh looks like it's already updated. I thought that wasn't supposed to happened until this PR was merged?

Also, I think this PR should be against master (I updated it), and it looks like it might need some rebasing.

**Comment by maebert** [comment]
@wren Oh damn, I think I accidentally deployed that form my branch while testing :-/ 

Rebased against master, should be good to go now.

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      - index.html
+extra_css:
+    - https://fonts.googleapis.com/css?family=Open+Sans:300,600
+    - assets/theme.css
+    - assets/highlight.css
+markdown_extensions:
+  - admonition
+repo_url: https://github.com/jrnl-org/jrnl/
+site_author: Manuel Ebert
+site_description: Never Worry about Money Again.
```
Comment (wren) [inline mkdocs.yml]: ?

**Pull request merged**